<?xml version="1.0" ?>
<robot name="LD90">

    <webots>
        <plugin type="webots_ros2_control::Ros2Control"/>

        <device reference="main_lidar" type="Lidar">
            <ros>
                <enabled>true</enabled>
                <updateRate>5</updateRate>
                <topicName>/scan</topicName>
                <alwaysOn>true</alwaysOn>
                <frameName>main_lidar</frameName>
            </ros>

        </device>
        <device reference="camera_realsense_depth_01" type="Lidar">
            <ros>
                <enabled>true</enabled>
                <updateRate>5</updateRate>
                <topicName>/scan_01</topicName>
                <alwaysOn>true</alwaysOn>
                <frameName>sub_lidar_01</frameName>
            </ros>
        </device>
        <device reference="camera_realsense_depth_02" type="Lidar">
            <ros>
                <enabled>true</enabled>
                <updateRate>5</updateRate>
                <topicName>/scan_02</topicName>
                <alwaysOn>true</alwaysOn>
                <frameName>sub_lidar_02</frameName>
            </ros>
        </device>


        <device reference="rear_sonar_1" type="DistanceSensor">
            <ros>
                <enabled>true</enabled>
                <updateRate>5</updateRate>
                <topicName>/ds1</topicName>
                <alwaysOn>true</alwaysOn>
                <frameName>rear_sonar_1</frameName>
            </ros>
        </device>
        <device reference="rear_sonar_2" type="DistanceSensor">
            <ros>
                <enabled>true</enabled>
                <updateRate>5</updateRate>
                <topicName>/ds2</topicName>
                <alwaysOn>true</alwaysOn>
                <frameName>rear_sonar_2</frameName>
            </ros>
        </device>
        <device reference="rear_sonar_3" type="DistanceSensor">
            <ros>
                <enabled>true</enabled>
                <updateRate>5</updateRate>
                <topicName>/ds3</topicName>
                <alwaysOn>true</alwaysOn>
                <frameName>rear_sonar_3</frameName>
            </ros>
        </device>
        <device reference="rear_sonar_4" type="DistanceSensor">
            <ros>
                <enabled>true</enabled>
                <updateRate>5</updateRate>
                <topicName>/ds4</topicName>
                <alwaysOn>true</alwaysOn>
                <frameName>rear_sonar_4</frameName>
            </ros>
        </device>
         <device reference="oakd_rgb_camera" type="Camera">
            <ros>
                <enabled>true</enabled>
                <updateRate>30</updateRate>
                <topicName>/oakd_rgb_camera/image_raw</topicName>
                <alwaysOn>true</alwaysOn>
                <frameName>oakd_rgb_camera_frame</frameName>
            </ros>
        </device>


    </webots>

    <ros2_control name="WebotsControl" type="system">
        <hardware>
            <plugin>webots_ros2_control::Ros2ControlSystem</plugin>
        </hardware>
        <joint name="left_wheel_joint">
            <state_interface name="position"/>
            <command_interface name="velocity"/>            
        </joint>
        <joint name="right_wheel_joint">
            <state_interface name="position"/>
            <command_interface name="velocity"/>            
        </joint>        
    </ros2_control>


    <link name="oakd_rgb_camera_link">
        <inertial>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <mass value="0.010"/>
            <inertia ixx="1.666667e-07" ixy="0.0" ixz="0.0" iyy="1.666667e-07" iyz="0.0" izz="1.666667e-07"/>
        </inertial>
        <visual>
            <geometry>
                <box size="0.05 0.05 0.05"/>
            </geometry>
        </visual>
        <collision>
            <geometry>
                <box size="0.05 0.05 0.05"/>
            </geometry>
        </collision>
    </link>

    <joint name="oakd_rgb_camera_joint" type="fixed">
        <parent link="base_link"/>
        <child link="oakd_rgb_camera_link"/>
        <origin xyz="0.058400 0.000000 0.096760" rpy="-0.577350 0.577350 -0.577350"/>
    </joint>
    
</robot>